bvm/ballerina-core/src/main/java/org/ballerinalang/model/util/ListUtils.java

```diff
@@ -47,66 +42,66 @@
+            return bRefValueArray.getRefValue(index);
         }
 
         switch (((BArrayType) array.getType()).getElementType().getTag()) {
             case TypeTags.BOOLEAN_TAG:
-                BBooleanArray bBooleanArray = (BBooleanArray) array;
-                int i = bBooleanArray.get(index);
+                BValueArray bBooleanArray = (BValueArray) array;
+                int i = bBooleanArray.getBoolean(index);
                 return i == 0 ? new BBoolean(false) : new BBoolean(true);
             case TypeTags.BYTE_TAG:
-                BByteArray bByteArray = (BByteArray) array;
-                return new BByte(bByteArray.get(index));
+                BValueArray bByteArray = (BValueArray) array;
+                return new BByte(bByteArray.getByte(index));
             case TypeTags.FLOAT_TAG:
-                BFloatArray bFloatArray = (BFloatArray) array;
-                return new BFloat(bFloatArray.get(index));
+                BValueArray bFloatArray = (BValueArray) array;
+                return new BFloat(bFloatArray.getFloat(index));
             case TypeTags.DECIMAL_TAG:
-                BRefValueArray bDecimalArray = (BRefValueArray) array;
-                return new BDecimal(new BigDecimal(bDecimalArray.get(index).stringValue()));
+                BValueArray bDecimalArray = (BValueArray) array;
+                return new BDecimal(new BigDecimal(bDecimalArray.getRefValue(index).stringValue()));
             case TypeTags.INT_TAG:
-                BIntArray bIntArray = (BIntArray) array;
-                return new BInteger(bIntArray.get(index));
+                BValueArray bIntArray = (BValueArray) array;
+                return new BInteger(bIntArray.getInt(index));
             case TypeTags.STRING_TAG:
-                BStringArray bStringArray = (BStringArray) array;
-                return new BString(bStringArray.get(index));
+                BValueArray bStringArray = (BValueArray) array;
+                return new BString(bStringArray.getString(index));
             default:
-                BRefValueArray bRefValueArray = (BRefValueArray) array;
-                return bRefValueArray.get(index);
+                BValueArray bRefValueArray = (BValueArray) array;
+                return bRefValueArray.getRefValue(index);
         }
     }
 
     public static void execListAddOperation(BNewArray array, long index, BRefType refType) {
         if (array.getType().getTag() != TypeTags.ARRAY_TAG) {
-            BRefValueArray bRefValueArray = (BRefValueArray) array;
+            BValueArray bRefValueArray = (BValueArray) array;
             bRefValueArray.add(index, refType);
             return;
         }
 
         switch (((BArrayType) array.getType()).getElementType().getTag()) {
             case TypeTags.BOOLEAN_TAG:
-                BBooleanArray bBooleanArray = (BBooleanArray) array;
+                BValueArray bBooleanArray = (BValueArray) array;
```

Comment:
can move this line out of the switch case.

Reply:
Done.